cfdm/data/mixin/arraymixin.py

```diff
@@ -137,6 +139,63 @@ def get_compression_type(self):
         """
         return self._get_component("compression_type", "")
 
+    def get_filename(self, default=AttributeError()):
+        """The name of the file containing the array.
+
+        If there are multiple files then an `AttributeError` is
+        raised.
```

Comment:
```suggestion
        raised by default.
```

Just to avoid any confusion RE the existence of the `default` parameter.